docs/source/tree.rst

```diff
@@ -223,13 +223,11 @@ allowed.
 Numeric literals
 ----------------
 
-While it is possible to store arbitrary-sized integers as literals in
-YAML, not all programming languages and YAML libraries are able to
-read them.  Therefore, to ensure portability, all numeric literals in
-the tree must assume that the reader has no more precision than that
-of a 64-bit double precision floating point number: 52-bits of
-precision.  Therefore, ASDF libraries should refuse to write files
-containing integers that are larger than 52-bits.
+To store arbitrary-sized integers as literals in YAML and for YAML
+libraries, and the ASDF API, to properly read and interpret them,
+integers represented as string literals in the tree must be no more
```

Comment:
This:
```
integers represented as string literals in the tree must be no more than 64-bits.
```
seems redundant with the last sentence:
```
For integers, the string literal must be within the range represented by int64 types.
```
```

Reply:
Instead of having a compound dependent clause, perhaps something like this is clearer:

> Integers represented as string literals in the ASDF tree must be no more than 64-bits. This limit exists because the YAML libraries used to interpret them must be able to round-trip the values exactly.  This is to ensure portability.

Reply:
Qualifying 64-bits with `int64` isn't redundant because 64-bit integers can be interpreted in two different ways, `uint64`, and `int64`.  Also, I didn't run into any problems with YAML parsing for very large numbers in generic arrays, but arose when putting them in `ndarrays` in `numpy`, which are valid for only specific `dtypes`.

Reply:
Oh, I see, does "int64" signify "signed 64-bit integer"?  Is that a standard term of some kind?  I'd be inclined to spell out the "signed" so dummies like me don't get confused.

Reply:
Yes.  An unsigned 64-bit integer is termed `uint64`, whereas a signed 64-bit integer is termed `int64`.  In the documents it might be good to say "signed 64-bit integer" as well as `int64`.